llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1536,20 +1492,24 @@ private SelectHostResult selectHost(TaskInfo request) {
       }
 
       // requested host is still alive but cannot accept task, pick the next available host in consistent order
-      for (int i = 0; i < allNodes.size(); i++) {
-        NodeInfo nodeInfo = allNodes.get((i + requestedHostIdx + 1) % allNodes.size());
-        // next node in consistent order died or does not have free slots, rollover to next
-        if (nodeInfo == null || !nodeInfo.canAcceptTask()) {
-          continue;
-        } else {
-          if (LOG.isDebugEnabled()) {
-            LOG.debug("Assigning {} in consistent order when looking for first requested host, from #hosts={},"
-                + " requestedHosts={}", nodeInfo.toShortString(), allNodes.size(),
-              ((requestedHosts == null || requestedHosts.length == 0) ? "null" :
-                requestedHostsDebugStr));
+      if (!activeNodesWithFreeSlots.isEmpty()) {
+        NodeInfo nextSlot = null;
+        boolean found = false;
+        for (Entry<String, List<NodeInfo>> entry : availableHostMap.entrySet()) {
+          if (found && !entry.getValue().isEmpty()) {
+            nextSlot = entry.getValue().iterator().next();
+            break;
           }
-          return new SelectHostResult(nodeInfo);
+          if (entry.getKey().equals(firstRequestedHost)) found = true;
+        }
+        // rollover
+        if (nextSlot == null) nextSlot = activeNodesWithFreeSlots.stream().findFirst().get();
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Assigning {} in consistent order when looking for first requested host, from #hosts={},"
```

Comment:
Make this info level as well. All the other assigning {} logs are info.

Reply:
Sure fixed